src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,571 @@
+      const totalSmartWalletPLRTokens = getTotal(accountBalanceWithSupportedChains, 'smartWallet');
+
+      setTotalKeyBasedPLRTokens(totalKeyBasedPLRTokens);
+      setTotalSmartWalletPLRTokens(totalSmartWalletPLRTokens);
+      setAccounts(accountBalanceWithSupportedChains);
+    } catch (e) {}
+  };
+
+  useEffect(() => {
+    // Fetch token balance for all the chains
+    fetchAllAccountBalances();
+  }, []);
+
+  const updateAvailableOffers = useCallback<() => Promise<ExchangeOffer[] | undefined>>(
+    debounce(async () => {
+      // there is a race condition here
+      if (multiCallData && fixed) {
+        return;
+      }
+      setSelectedOffer(null);
+      setAvailableOffers([]);
+
+      if (!sdk || !selectedFromAsset || !amount || !selectedFromNetwork?.chainId || !isValidAmount(amount)) return;
+
+      setIsLoadingAvailableOffers(true);
+
+      try {
+        // needed computed account address before calling getExchangeOffers
+        if (!accountAddress) await sdk.computeContractAccount();
+        const offers = await sdk.getExchangeOffers({
+          fromChainId: selectedFromAsset.chainId,
+          fromAmount: ethers.utils.parseUnits(amount, selectedFromAsset.decimals),
+          toTokenAddress: hasEnoughPLR ? plrDaoMemberNFT.address : plrDaoAsset.address,
+          fromTokenAddress: selectedFromAsset.address,
+        });
+        return offers;
+      } catch (e) {
+        setTransactionBlockFieldValidationError(transactionBlockId, 'offer', 'Cannot fetch offers');
+      }
+    }, 200),
+    [sdk, selectedFromAsset, amount, selectedFromNetwork, accountAddress, selectedAccountType]
+  );
+
+  useEffect(() => {
+    // Fetch a list of NFTs for the account to check if the user is existing member of PLR Dao.
+    try {
+      if (accountAddress && providerAddress) {
+        sdk
+          ?.getNftList({
+            account: accountAddress || providerAddress,
+          })
+          .then((output: NftList) => {
+            let hasNFTContractAddress = output?.items?.filter((nft) => nft.contractAddress === plrDaoMemberNFT.address);
+            if (hasNFTContractAddress?.length) {
+              setIsNFTMember(true);
+            }
+          });
+      }
+    } catch (error) {}
+  }, [sdk, accountAddress, providerAddress]);
+
+  useEffect(() => {
+    // this will ensure that the old data won't replace the new one
+    let active = true;
+    const update = async () => {
+      try {
+        const offers = await updateAvailableOffers();
+        if (active && offers) {
+          setAvailableOffers(offers);
+          setIsLoadingAvailableOffers(false);
+          if (!offers.length) return;
+          const bestOffer: any = offers?.find((offer) => offer.provider === swapServiceIdToDetails['Lifi'].title);
+          const selectedOffer = bestOffer?.provider ? bestOffer : offers[0];
+          setSelectedOffer(mapOfferToOption(selectedOffer));
+        }
```

Comment:
consistency in your own code
```suggestion
        if (!active || !offers) return;
        setAvailableOffers(offers);
        setIsLoadingAvailableOffers(false);
        if (!offers.length) return;
        const bestOffer: any = offers?.find((offer) => offer.provider === swapServiceIdToDetails['Lifi'].title);
        const selectedOffer = bestOffer?.provider ? bestOffer : offers[0];
        setSelectedOffer(mapOfferToOption(selectedOffer));
```